app/src/js/reducers/granules.js

```diff
@@ -65,141 +66,157 @@ export const initialState = {
+    const { id } = action;
+    set(state, ['executed', id, 'status'], 'success');
+    set(state, ['executed', id, 'error'], null);
+  },
+  [GRANULE_APPLYWORKFLOW_INFLIGHT]: (state, action) => {
+    const { id } = action;
+    set(state, ['executed', id, 'status'], 'inflight');
+  },
+  [GRANULE_APPLYWORKFLOW_ERROR]: (state, action) => {
+    const { id } = action;
+    set(state, ['executed', id, 'status'], 'error');
+    set(state, ['executed', id, 'error'], action.error);
+  },
+
+  [GRANULE_REMOVE]: (state, action) => {
+    const { id } = action;
+    set(state, ['removed', id, 'status'], 'success');
+    set(state, ['removed', id, 'error'], null);
+  },
+  [GRANULE_REMOVE_INFLIGHT]: (state, action) => {
+    const { id } = action;
+    set(state, ['removed', id, 'status'], 'inflight');
+  },
+  [GRANULE_REMOVE_ERROR]: (state, action) => {
+    const { id } = action;
+    set(state, ['removed', id, 'status'], 'error');
+    set(state, ['removed', id, 'error'], action.error);
+  },
+
+  [BULK_GRANULE]: (state, action) => {
+    const { data, config } = action;
+    set(state, ['bulk', config.requestId, 'data'], data);
+    set(state, ['bulk', config.requestId, 'status'], 'success');
+    set(state, ['bulk', config.requestId, 'error'], null);
+  },
+  [BULK_GRANULE_INFLIGHT]: (state, action) => {
+    const { config } = action;
+    set(state, ['bulk', config.requestId, 'status'], 'inflight');
+  },
+  [BULK_GRANULE_ERROR]: (state, action) => {
+    const { config } = action;
+    set(state, ['bulk', config.requestId, 'status'], 'error');
+    set(state, ['bulk', config.requestId, 'error'], action.error);
+  },
+
+  [GRANULE_DELETE]: (state, action) => {
+    const { id } = action;
+    set(state, ['deleted', id, 'status'], 'success');
+    set(state, ['deleted', id, 'error'], null);
+  },
+  [GRANULE_DELETE_INFLIGHT]: (state, action) => {
+    const { id } = action;
+    set(state, ['deleted', id, 'status'], 'inflight');
+  },
+  [GRANULE_DELETE_ERROR]: (state, action) => {
+    const { id } = action;
+    set(state, ['deleted', id, 'status'], 'error');
+    set(state, ['deleted', id, 'error'], action.error);
+  },
+
+  [SEARCH_GRANULES]: (state, action) => {
+    set(state, ['list', 'params', 'prefix'], action.prefix);
+  },
+  [CLEAR_GRANULES_SEARCH]: (state, action) => {
+    set(state, ['list', 'params', 'prefix'], null);
+  },
+
+  [FILTER_GRANULES]: (state, action) => {
+    set(state, ['list', 'params', action.param.key], action.param.value);
+  },
+  [CLEAR_GRANULES_FILTER]: (state, action) => {
+    set(state, ['list', 'params', action.paramKey], null);
+  },
+
+  [OPTIONS_COLLECTIONNAME]: (state, action) => {
+    const { data } = action;
+    // Map the list response to an object with key-value pairs like:
+    // displayValue: optionElementValue
+    const options = data.results.reduce((obj, d) => {
+      const { name, version } = d;
+      obj[`${name} ${version}`] = `${name}___${version}`;
+      return obj;
+    }, {});
```

Comment:
nit: replace `${name}___${version}` with `getCollectionId(d)` (from `app/src/js/utils/format.js`)

or, if you're feeling a bit more ambitious, change the highlighted block to this (although this is arguably no more readable, so feel free to ignore my digression):
```
const options = action.data.results.reduce(
  (obj, { name, version }) =>
    Object.assign(obj, {
      [`${name} ${version}`]: getCollectionId({ name, version })
    }),
  {}
);
```